### DISCUSSION TITLE
Automation in DAW : Reaper ...

### DISCUSSION THREAD

**Discussion by subderground** [DISTRHO/Cardinal#235] [Q&A]
Hi, 
The automation of Cardinal in DAW (Reaper) seems to not working. On "automation write mode", when I modify a parameter (ex Cutoff, VCA, or...), nothing is recognised by the DAW... so no automation. 
I've tried the VST and VST3 versions of Cardinal 22.05.

Nb: No problem with the old VeeSeeVST, the automation works well like all VST / VST3 plugins. 

Can you test that on your DAW ? If it's not working... Could you fix that please ?

Many thanks.

**Comment by falkTX** [comment]
Nothing to fix, this is just not how Cardinal works.
You cannot map the plugin-host-exposed parameters to random stuff inside Cardinal.

For now you can use the "Host Params" module to receive parameter data from the host.
Later on there will be a "Host Param-Map", similar to "Host MIDI-Map", but this is not done yet.

**Reply by subderground** [reply]
Ok, thanks. It's not a good news for me. VeeCeeVSt and Vcv pro are able to do that, I was thinking Cardinal too. For me, the possibily to simply automate a parameter (aslo if there is no '' virtual jack'' cv/gate input in the module knob/fader/switch) is essential to make a track. I just hope this will comes in the futur. Have a nice day.

**Reply by falkTX** [reply]
tbh I find it very weird that, with cables everywhere, we skip on that entirely for plugin/host parameter automation.
having no way to see what is currently mapped to what seems very bad to me.
I never used Rack Pro, so do not know how it is handled there, but anyway for Cardinal as I said at least a dedicated parameter will be used so the mapping can be shown there.

**Comment by subderground** [comment]
Yes, perhaps, if you have time, you could test '' veeseevst rack'', and check the code on the github. It works fine with daw automations :-)
Btw Cardinal is a really cool project. I hope it will grow :-) Thanks for this.

**Reply by falkTX** [reply]
I have seen the older vst yes. and I dislike how it does things.
Whenever I get to this (if no one does it sooner) I want it to be more transparent and less confusing.

**Comment by offthesky** [comment]
@subderground - here's the fastest way i found to do it in reaper: 
[cardinal using host midi cc map (screenshot)](https://user-images.githubusercontent.com/95579081/169317847-f7e468eb-fa3b-42b2-92a4-229e96e9023e.png)

also i had put in a request before for something that would also make this much easier. perhaps maybe we will see something like it in cardinal down the road :)
https://github.com/DISTRHO/Cardinal/issues/110

**Reply by subderground** [reply]
Ok, that's a good alternative to automate Cardinal... not the fastest, but it 's always cool to have a workaround. :-) 
Thanks.

**Reply by subderground** [reply]
Sorry @offthesky Your workaround to automate Cardinal "Cutoff Filter" with "cc midi host" doesn't work here for me. I've just exactly followed your instructions (same chain with CClfo before Cardinal, same midi channel, moving the parameters, ect...) I have  also tested with ReaControlMIDI (I use it to control Hardware) instead of CClfo, and even directly with midi CC from hardware. :-( It seems like "Host midi cc map module" doesn't recieve the CC infos.  A little red square is on the parameter "assigned" (Filter Cutoff) on the Cardinal GUI module.
But unlike you, I don't have "CC 0x" before the parameter (Cutoff) in Cardinal "Host midi cc map module".   
PS: It's the same with "Cardinal 18ch" & "Cardinal synth" VST3 (midi notes are recognized, so I don't know why midi cc not)

**EDIT: I found it !! It's a bug of the VST3 version, on VST2 your workaround is good.**


Thanks for your help.

**Reply by falkTX** [reply]
Cardinal VST3 does not do MIDI CC or pitchbend at the moment, it is complicated to explain.. vst3 is very weird when it comes to MIDI

**Reply by subderground** [reply]
No problem, I'll use the vst2 instead. Thanks.

**Reply by subderground** [reply]
Now (22.06 version), everything is ok with booth VST2 and VST3 versions (64bit Reaper Windows11). Thanks for this great update ! Nice work !
